LogStream.computeNext throws an exception EOFException despite of working container

My Docker container has dockered the sipp application. When I run it from docker command tool, 

`docker run -t localhost:5000/application:v2 /data/go.sh`

I can see the continuous output.

But when I run the java code


```
@Test
    public void testMe() throws DockerException, InterruptedException {
        try (DockerClient docker = DefaultDockerClient.builder()
                .uri(URI.create("http://docker.my.com:2375"))
                .build()) {
 
            ContainerConfig containerConfig = ContainerConfig.builder()
                    .image("localhost:5000/application:v2")
                    .attachStdout(true)
                    .tty(true)
                    .cmd("/data/go.sh")
                    .build();

            ContainerCreation containerCreation = docker.createContainer(containerConfig);
            docker.startContainer(containerCreation.id());
            LogStream logStream = docker.logs(containerCreation.id(), DockerClient.LogsParameter.STDOUT);

            while (logStream.hasNext()) {
                System.out.println("LINE: " + logStream.next());
            }
        }
    }
```

it performs the running of this container. It works well at docker

`[mami@me ~]$ docker ps
CONTAINER ID        IMAGE                             COMMAND                  CREATED             STATUS              PORTS                          NAMES
ccf9eb647afd        localhost:5000/application:v2   "/data/go.sh"            9 seconds ago       Up 8 seconds           modest_mclean
`

but my test throws the exception

```
java.lang.RuntimeException: java.io.EOFException: reached end of stream after reading 67 bytes; 1667330080 bytes expected
        at com.google.common.base.Throwables.propagate(Throwables.java:160)
        at com.spotify.docker.client.LogStream.computeNext(LogStream.java:69)
        at com.spotify.docker.client.LogStream.computeNext(LogStream.java:43)
        at com.google.common.collect.AbstractIterator.tryToComputeNext(AbstractIterator.java:143)
        at com.google.common.collect.AbstractIterator.hasNext(AbstractIterator.java:138)
        at com.myapplication.tests.performance.DockerTest.testMe(DockerTest.java:34)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:45)
        at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:15)
        at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:42)
        at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:20)
        at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:263)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:68)
        at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:47)
        at org.junit.runners.ParentRunner$3.run(ParentRunner.java:231)
        at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:60)
        at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:229)
        at org.junit.runners.ParentRunner.access$000(ParentRunner.java:50)
        at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:222)
        at org.junit.runners.ParentRunner.run(ParentRunner.java:300)
        at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
        at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
        at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:483)
        at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
        at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
        at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
        at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
        at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
Caused by: java.io.EOFException: reached end of stream after reading 67 bytes; 1667330080 bytes expected
        at com.google.common.io.ByteStreams.readFully(ByteStreams.java:787)
        at com.google.common.io.ByteStreams.readFully(ByteStreams.java:766)
        at com.spotify.docker.client.LogReader.nextMessage(LogReader.java:69)
        at com.spotify.docker.client.LogStream.computeNext(LogStream.java:67)
        ... 33 more
```


When I attach to this working container via "docker attach" I can see the appending output.

How can I read the output with LogsStream here?

--- a/src/main/java/com/spotify/docker/client/LogReader.java
+++ b/src/main/java/com/spotify/docker/client/LogReader.java
@@ -18,6 +18,7 @@
 package com.spotify.docker.client;
 
 import com.google.common.io.ByteStreams;
+import com.spotify.docker.client.LogMessage.Stream;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
@@ -45,6 +46,7 @@ public class LogReader implements Closeable {
   }
 
   public LogMessage nextMessage() throws IOException {
+    stream.mark(HEADER_SIZE);
 
     // Read header
     final byte[] headerBytes = new byte[HEADER_SIZE];
@@ -56,12 +58,21 @@ public class LogReader implements Closeable {
       throw new EOFException();
     }
     final ByteBuffer header = ByteBuffer.wrap(headerBytes);
-    final int streamId = header.get();
-    header.position(FRAME_SIZE_OFFSET);
-    final int frameSize = header.getInt();
+    int streamId = header.get();
+    final int idZ = header.getInt(0);
 
     // Read frame
-    final byte[] frame = new byte[frameSize];
+    final byte[] frame;
+    // Header format is : {STREAM_TYPE, 0, 0, 0, SIZE1, SIZE2, SIZE3, SIZE4}
+    if (idZ == 0 || idZ == 0x01000000 || idZ == 0x02000000) {
+      header.position(FRAME_SIZE_OFFSET);
+      final int frameSize = header.getInt();
+      frame = new byte[frameSize];
+    } else {
+      stream.reset();
+      streamId = Stream.STDOUT.id();
+      frame = new byte[stream.available()];
+    }
     ByteStreams.readFully(stream, frame);
     return new LogMessage(streamId, ByteBuffer.wrap(frame));
   }

--- a/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
+++ b/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
@@ -1823,6 +1823,30 @@ public class DefaultDockerClientTest {
     assertThat(logs, not(containsString("This message was printed too late")));
   }
 
+  @Test
+  public void testLogsTty() throws DockerException, InterruptedException {
+    String container = randomName();
+    ContainerConfig containerConfig = ContainerConfig.builder()
+        .image(BUSYBOX_LATEST)
+        .attachStdout(true)
+        .tty(true)
+        .cmd("sh", "-c", "ls")
+        .build();
+
+    sut.createContainer(containerConfig, container);
+    sut.startContainer(container);
+    LogStream logStream = sut.logs(container, DockerClient.LogsParam.stdout());
+
+    while (logStream.hasNext()) {
+      String line = UTF_8.decode(logStream.next().content()).toString();
+      log.info(line);
+    }
+    sut.waitContainer(container);
+    final ContainerInfo info = sut.inspectContainer(container);
+    assertThat(info.state().running(), is(false));
+    assertThat(info.state().exitCode(), is(0));
+  }
+
   @Test(expected = ContainerNotFoundException.class)
   public void testStartBadContainer() throws Exception {
     sut.startContainer(randomName());

